lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,360 @@
+            if (!pdu.checkTagIntegrity(tag)) {
+                this._sockets[index]._events.emit(
+                    'getResponse', errors.InternalError);
+            }
+            this._sockets[index]._events.emit('getResponse', null, chunk);
+        } else {
+            this._chunkTab.push(chunk);
+        }
+    }
+
+    _endGet(index) {
+        let chunk = Buffer.allocUnsafe(0);
+        if (this._count !== this._sockets[index]._chunkSize) {
+            chunk = this._sockets[index].sock.read();
+            this._chunkTab.push(chunk);
+            this._count += chunk.length;
+        }
+        if (this._count === this._sockets[index]._chunkSize) {
+            this._sockets[index]._events.emit(
+                'getResponse', null, Buffer.concat(this._chunkTab));
+        }
+    }
+
+    _newSocket() {
+        const socket = new net.Socket().pause();
+        socket.setNoDelay();
+        socket.setKeepAlive(true);
+        socket.unref();
+        return socket;
+    }
+
+    _propError(pdu) {
+        const statusCode = pdu.getStatusCode();
+        if (statusCode === kineticlib.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        if (statusCode !== kineticlib.errors.SUCCESS) {
+            return pdu.getErrorMessage();
+        }
+        return undefined;
+    }
+
+    /**
+     * set the sequence of the request.
+     *
+     * @param {number} index - the index of the socket.
+     * @param {number} sequence - the sequence to be set.
+     * @returns {Kinetic} - To allow for a functional style.
+     */
+    setSequence(index, sequence) {
+        if (sequence >= Number.MAX_VALUE) {
+            const temp = this._sockets[index].sock;
+            this._sockets[index].sock = this._newSocket();
+            this._sockets[index].sock.on('connect', () => {
+                this._bindReadable(this._sockets[index], index);
+            }).connect(this._sockets[index].drive);
+            temp.destroy();
+            this._sockets[index].sequence = 0;
+        } else {
+            this._sequence = sequence;
+            this._sockets[index].sequence = sequence;
+        }
+        return this;
+    }
+
+    /**
+     * get the pool of socket.
+     *
+     * @returns {Kinetic._sockets} - the pool of socket.
+     */
+    getSockets() {
+        return this._sockets;
+    }
+
+    /**
+     * get the socket at the index.
+     *
+     * @param {number} index - the index of the socket.
+     * @returns {Object} - the socket.
+     */
+    getSocket(index) {
+        return this._sockets[index];
+    }
+
+    /**
+     * get index of the socket.
+     *
+     * @param {string} host - the host of the socket.
+     * @param {string} port - the port of the socket.
+     * @returns {number|object} - the index of the socket or null if it does not
+     *                            exist.
+     */
+    getSocketIndex(host, port) {
+        this._sockets.forEach((socket, index) => {
```

Comment:
If you want to return the index, you'll need to use the `findIndex()` array method instead.